projects/extension/ai/semantic_catalog.py

```diff
@@ -0,0 +1,433 @@
+        )
+        existing_columns = dict()
+        for r in result:
+            existing_columns[r["objnames"][-1]] = True
+
+        plan = plpy.prepare(
+            "select ai.set_column_description($1, $2, $3)", ["text", "text", "text"]
+        )
+        for column in columns:
+            exists = column["name"] in existing_columns
+            if not exists or overwrite:
+                plpy.debug(
+                    f"set description for {column['name']} (existing={exists}, overwrite={overwrite})"
+                )
+                plpy.execute(plan, [relation, column["name"], column["description"]])
+    for column in columns:
+        yield column["name"], column["description"]
+
+
+def generate_function_description(
+    plpy,
+    fn: str,
+    catalog_name: str,
+    config: dict | None,
+    save: bool,
+    overwrite: bool,
+) -> Generator[GeneratedDescription, None, None]:
+    parsed_config = get_parsed_config(plpy, catalog_name, config)
+    description = "None"
+    system_prompt = """
+    You are a SQL expert generates natural language descriptions of functions in a database.
+
+    The descriptions that you generate should be a concise single sentence.
+    """
+
+    message_content = f"""
+    Given the following function, generate a natural language description of it:
+
+    {get_obj_description(plpy, fn=fn)}
+    """
+
+    tools = [
+        {
+            "name": "generate_description",
+            "description": "Record description of function in well-formed JSON",
+            "input_schema": {
+                "type": "object",
+                "properties": {
+                    "description": {
+                        "type": "string",
+                        "description": "Description of the function",
+                    }
+                },
+            },
+        }
+    ]
+
+    provider = parsed_config.get("provider", None)
+    if provider == "anthropic":
+        model = parsed_config.get("model", "claude-3-5-sonnet-latest")
+        messages = [{"role": "user", "content": message_content}]
+
+        plan = plpy.prepare(
+            """
+                select ai.anthropic_generate(
+                    $1
+                    , $2
+                    , system_prompt => $3
+                    , tools => $4
+                    , tool_choice => '{{"type": "tool", "name": "generate_description"}}'::jsonb
+                )
+            """,
+            ["text", "jsonb", "text", "jsonb"],
+        )
+        result = plpy.execute(
+            plan, [model, json.dumps(messages), system_prompt, json.dumps(tools)]
+        )
+        response = json.loads(result[0]["anthropic_generate"])
+        description = response["content"][0]["input"]["description"]
+    elif provider == "ollama":
+        pass
```

Comment:
raise exception

Reply:
Done in 27853696d476c88c7c8ae47fa003081e711dc8ed